packages/react-instantsearch-dom/src/components/VoiceSearch.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react';
+import { translatable, Translate } from 'react-instantsearch-core';
+import { createClassNames } from '../core/utils';
+import voiceSearchHelper, {
+  VoiceSearchHelper,
+  VoiceListeningState,
+} from '../lib/voiceSearchHelper';
+const cx = createClassNames('VoiceSearch');
+
+type InnerComponentProps = {
+  status: string;
+  errorCode?: string;
+  isListening: boolean;
+  transcript?: string;
+  isSpeechFinal?: boolean;
```

Comment:
My reason behind this is here: https://github.com/algolia/angular-instantsearch/pull/530#discussion_r283756680
What do you think?

Reply:
I would have stick with only the boolean. If the value have to carry another meaning it probably means that the boolean is not the appropriate type. The information that the recognition didn't start could be provided by another value (actually `isListening` could probably used for that), or use an enum rather than a boolean. Anyway it doesn't have a huge impact.

Reply:
It makes sense. A boolean variable containing three meanings shouldn't be a boolean. Now that there's no strong reason to store three meanings, I'll just change it to non-optional boolean. I'm going to do this change to IS.js and all the flavours.

Reply:
https://github.com/algolia/react-instantsearch/pull/2316/commits/5fa8fa0163ac752cf78171118e6f0de1e66942f4